### PR TITLE
#600 JavaDocs for Props farm

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #600

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#923](https://github.com/zerocracy/farm/issues/923) is now in scope, role is `REV`

File: src/main/java/com/zerocracy/farm/props/package-info.java [hunk 1]
```diff
@@ -16,7 +16,15 @@
  */
 
 /**
- * Props farm.
+ * Props farm. Classes in this package are used to make sure that the Farm
+ * reflects the _props.xml file containing properties such as sensitive
+ * credentials used in PMO project.
+ *
+ * It is used in the decoration chain, in the ctor SmartFarm(farm), which is the
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/farm/props/package-info.java]: @amihaiemil let's use a Javadoc link (see [here](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDDIECH)) to refer to the exact `SmartFarm` constructor.

File: src/main/java/com/zerocracy/farm/props/package-info.java [hunk 2]
```diff
@@ -16,7 +16,15 @@
...
+ * Farm entry point used in Main.exec.
+ *
+ * Also, PropsFarm may be used directly in tests where PMO is needed,
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/farm/props/package-info.java]: @amihaiemil same here, let's use a [Javadoc link](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDDIECH) to refer to `PropsFarm`.

File: src/main/java/com/zerocracy/farm/props/PropsFarm.java [hunk 3]
```diff
@@ -27,7 +27,8 @@
 import org.xembly.Directives;
 
 /**
- * Props farm.
+ * Props farm. This Farm decorator will make sure all the Projects have their
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/farm/props/PropsFarm.java]: @amihaiemil let's add a [Javadoc link](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDDIECH) to `Farm` type here.

**Comment by amihaiemil** [comment]
@carlosmiranda done

**Comment by amihaiemil** [comment]
@yegor256 ping, can we merge this?

File: src/main/java/com/zerocracy/farm/props/package-info.java [hunk 4]
```diff
@@ -16,7 +16,16 @@
...
 
 /**
- * Props farm.
+ * Props farm. Classes in this package are used to make sure that the
+ * {@link Farm} reflects the _props.xml file containing properties such as
+ * sensitive credentials used in PMO project.
+ *
+ * It is used in the decoration chain, in the constructor
+ * {@link #SmartFarm(Farm) SmartFarm(Farm)}, which is the Farm entry point
+ * used in Main.exec.
```
Comment (yegor256) [inline src/main/java/com/zerocracy/farm/props/package-info.java]: @amihaiemil why this method is not linked via `@link`?

File: src/main/java/com/zerocracy/farm/props/package-info.java [hunk 5]
```diff
@@ -16,7 +16,16 @@
  */
 
 /**
- * Props farm.
+ * Props farm. Classes in this package are used to make sure that the
+ * {@link Farm} reflects the _props.xml file containing properties such as
+ * sensitive credentials used in PMO project.
+ *
+ * It is used in the decoration chain, in the constructor
+ * {@link #SmartFarm(Farm) SmartFarm(Farm)}, which is the Farm entry point
```
Comment (yegor256) [inline src/main/java/com/zerocracy/farm/props/package-info.java]: @amihaiemil there is something wrong with this link

**Comment by yegor256** [comment]
@amihaiemil the formatting is a bit off. Make sure Javadoc starts with a single line, which ends with a period. Then, start them with `<p>`. See how it's formatted in `package-info.java` in `com.zerocracy.farm.guts`.

Also, make sure you use `{@code ...}` for code parts like `_props.xml`

**Comment by amihaiemil** [comment]
@yegor256 done

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by amihaiemil** [comment]
@yegor256 ping, don't forget to merge this :)

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#923](https://github.com/zerocracy/farm/issues/923) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Please add more details into the PR's description.

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review.
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka done :)

**Comment by ypshenychka** [comment]
@carlosmiranda @amihaiemil Thank you!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**